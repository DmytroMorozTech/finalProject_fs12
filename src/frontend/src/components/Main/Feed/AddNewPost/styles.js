import { makeStyles } from '@material-ui/core/styles'

export default makeStyles((theme) => ({

  title: {
    marginLeft: theme.spacing(4),
    color: theme.palette.grey[700],
    fontSize: theme.typography.h3.fontSize,
    fontWeight: theme.typography.h4.fontWeight,
    marginBottom: theme.spacing(3)
  },

  horizontalLine: {
    height: '1px',
    border: '0',
    margin: '0',
    backgroundColor: theme.palette.grey[100]
  },

  userInfo: {
    display: 'flex',
    flexDirection: 'row'
  },

  userAvatar: {
    width: theme.avatar.small,
    borderRadius: '50%'
  },

  buttonGroup: {
    paddingLeft: theme.spacing(2)
  },

  sharePost: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    borderRadius: theme.shape.large,
    cursor: 'pointer',
    color: theme.palette.grey[700],
    backgroundColor: theme.palette.common.white,
    border: theme.border.boldLight,
    paddingRight: theme.spacing(3),
    '&:hover': {
      backgroundColor: 'rgba(0,0,0,0.05)',
      border: theme.border.boldDark
    }
  },

  showedValidateMessage: {
    display: 'flex',
    justifyContent: 'space-between',
    color: theme.palette.error.main,
    alignItems: 'center',
    paddingLeft: theme.spacing(4),
    paddingRight: theme.spacing(4),
    paddingTop: theme.spacing(2)
  },

  hidden: {
    display: 'none'
  },

  validateMessage: {
    display: 'flex',
    fontSize: theme.typography.icons.small.fontSize,
    alignItems: 'center'
  },

  validateInfo: {
    display: 'flex',
    margin: theme.spacing(1),
    fontSize: theme.typography.h6.fontSize,
    fontWeight: theme.typography.fontWeightBold
  },

  worldIcon: {
    paddingLeft: theme.spacing(1),
    paddingRight: theme.spacing(1),
    paddingTop: theme.spacing(1),
    fontSize: theme.typography.icons.medium.fontSize,
    color: theme.palette.grey[800]
  },

  editor: {
    marginTop: theme.spacing(4)
  },

  shareButtons: {
    display: 'flex',
    flexDirection: 'row',
    color: theme.palette.grey[700],
    marginLeft: theme.spacing(2)
  },

  shareButton: {
    padding: theme.spacing(2),
    cursor: 'pointer',
    '&:hover': {
      backgroundColor: 'rgba(0,0,0,0.05)',
      borderRadius: '50%'
    }
  },

  verticalLine: {
    borderLeft: theme.border.simple,
    margin: theme.spacing(1)
  },

  previewImgWrapper: {
    maxWidth: '50%',
    maxHeight: '50%',
    position: 'relative'
  },

  previewImg: {
    width: '100%',
    borderRadius: theme.shape.medium
  },

  icons: {
    display: 'flex',
    cursor: 'pointer'
  },

  cross: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    width: theme.spacing(7),
    height: theme.spacing(7),
    fontSize: theme.typography.icons.medium.fontSize,
    color: theme.palette.common.white,
    backgroundColor: 'rgba(0,0,0,.5)',
    padding: theme.spacing(1),
    borderRadius: '50%',
    position: 'absolute',
    top: theme.spacing(2),
    right: theme.spacing(2),
    cursor: 'pointer'
  },

  crossForVideoNotification: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    width: theme.spacing(7),
    height: theme.spacing(7),
    fontSize: theme.typography.icons.medium.fontSize,
    color: theme.palette.common.white,
    backgroundColor: 'rgba(0,0,0,.5)',
    padding: theme.spacing(1),
    borderRadius: '50%',
    position: 'relative',
    cursor: 'pointer',
    marginLeft: theme.spacing(2)
  },

  removed: {
    display: 'none'
  },

  preloader: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'space-evenly',
    width: '100%',
    height: '35vh'
  },

  videoWasChosenNotification: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    position: 'relative'
  },

  nameVideo: {
    display: 'flex',
    flexDirection: 'column',
    width: '100%'
  }

}))
