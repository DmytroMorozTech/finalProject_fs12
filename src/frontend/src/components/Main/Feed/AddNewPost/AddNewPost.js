import React, { useState } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { withStyles } from '@material-ui/core/styles'
import toggleModalAction from '../../../../redux/Modal/modalActions'
import PublicIcon from '@material-ui/icons/Public'
import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown'
import InputBase from '@material-ui/core/InputBase'
import Typography from '@material-ui/core/Typography'
import MuiDialogContent from '@material-ui/core/DialogContent'
import MuiDialogActions from '@material-ui/core/DialogActions'
import Tooltip from '@material-ui/core/Tooltip'
import PhotoSizeSelectActualIcon from '@material-ui/icons/PhotoSizeSelectActual'
import YouTubeIcon from '@material-ui/icons/YouTube'
import EventNoteIcon from '@material-ui/icons/EventNote'
import styles from './styles'
import { activeUserSelector } from '../../../../redux/User/userSelector'
import { createNewPostAction } from '../../../../redux/Post/postActions'
import SharedButton from '../../../../shared/SharedButton/SharedButton'
import RemoveCircleIcon from '@material-ui/icons/RemoveCircle'
import { toast } from 'react-toastify'

const DialogContent = withStyles((theme) => ({
  root: {
    padding: theme.spacing(4),
    maxHeight: '450px',
    overflowY: 'auto'
  }
}))(MuiDialogContent)

const DialogActions = withStyles((theme) => ({
  root: {
    margin: 0,
    paddingRight: theme.spacing(4),
    paddingLeft: theme.spacing(0),
    justifyContent: 'space-between'
  }
}))(MuiDialogActions)

const LightTooltip = withStyles((theme) => ({
  tooltip: {
    backgroundColor: theme.palette.common.white,
    color: theme.palette.grey[800],
    boxShadow: theme.shadows[1],
    padding: theme.spacing(2),
    fontSize: theme.typography.h5.fontSize
  }
}))(Tooltip)

const AddNewPost = () => {
  const dispatch = useDispatch()
  const activeUser = useSelector(activeUserSelector)

  const [photoIsChosen, setPhotoIsChosen] = useState(false)
  const [selectedFile, setSelectedFile] = useState(null)
  const [imgIsUploading, setImgIsUploading] = useState(false)

  const onPostSubmitHandler = () => {
    dispatch(createNewPostAction({ text: postInputText }))
    dispatch(toggleModalAction())
  }

  const classes = styles()

  const [postInputText, setPostInputText] = React.useState('')

  const handlePostInputChange = e => {
    let postInputVal = e.currentTarget.value
    setPostInputText(postInputVal)
  }

  const numberCharacterToShowValidate = 3000
  const validateCount = (numberCharacterToShowValidate - postInputText.length)
  let btnIsDisabled = postInputText.length === 0 || postInputText.length > numberCharacterToShowValidate

  return (
    <div>
      <div className={classes.title}>
        Create a post
      </div>
      <hr className={classes.horizontalLine}/>
      <DialogContent>
        <div className={classes.userInfo}>
          <div>
            <img src={activeUser.avatarUrl} alt={'user avatar'} className={classes.userAvatar}/>
          </div>
          <div className={classes.buttonGroup}>
            <Typography variant="h5">
              {activeUser.fullName}
            </Typography>
            <button className={classes.sharePost}>
              <div className={classes.icons}>
                <PublicIcon fontSize="inherit"/>
              </div>
              <Typography variant="h4">
                Anyone
              </Typography>
              <div className={classes.icons}>
                <ArrowDropDownIcon fontSize="inherit"/>
              </div>
            </button>
          </div>
        </div>
        <InputBase
          placeholder="What do you want to talk about?"
          fullWidth={true}
          multiline={true}
          minRows={7}
          value={postInputText}
          onChange={handlePostInputChange}
          className={classes.editor}
        />
        {photoIsChosen ? <img src={URL.createObjectURL(selectedFile)}/> : null}
      </DialogContent>
      <div
        className={postInputText.length > numberCharacterToShowValidate ? classes.showedValidateMessage : classes.hidden}>
        <div className={classes.validateMessage}>
          <RemoveCircleIcon fontSize="inherit"/>
          <div className={classes.validateInfo}>You have exceeded the maximum character limit.</div>
        </div>
        <div className={classes.validateInfo}>
          {validateCount}
        </div>
      </div>
      <DialogActions>
        <div className={classes.shareButtons}>

          <LightTooltip title={`Add a photo`} placement={'top'}>
            <div className={classes.shareButton} >
              <label>
                <input
                  type="file"
                  id="file"
                  accept="image/*"
                  style={{display: 'none'}}
                  required
                  onChange={(event) => {
                    const file = event.target.files[0]
                    if (file.size > 10485760) {
                      toast.error('The size of image should not exceed 10MB')
                      return
                    }

                    setSelectedFile(file)
                    setPhotoIsChosen(true)
                  }}

                />
                <PhotoSizeSelectActualIcon/>
              </label>
            </div>
          </LightTooltip>

          <LightTooltip title={`Add a video`} placement={'top'}>
            <div className={classes.shareButton}>
              <YouTubeIcon/>
            </div>
          </LightTooltip>
          <LightTooltip title={`Add a document`} placement={'top'}>
            <div className={classes.shareButton}>
              <EventNoteIcon/>
            </div>
          </LightTooltip>
          <hr className={classes.verticalLine}/>
        </div>
        <SharedButton title="Post" disabled={btnIsDisabled} onClick={onPostSubmitHandler}/>
      </DialogActions>
    </div>
  )
}
export default AddNewPost
