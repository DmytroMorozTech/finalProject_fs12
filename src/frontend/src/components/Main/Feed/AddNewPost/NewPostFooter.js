import React, {useState} from 'react'
import {makeStyles} from '@material-ui/core/styles'
import PhotoSizeSelectActualIcon from '@material-ui/icons/PhotoSizeSelectActual'
import YouTubeIcon from '@material-ui/icons/YouTube'
import EventNoteIcon from '@material-ui/icons/EventNote'
import Tooltip from '@material-ui/core/Tooltip'

const useStyles = makeStyles(() => ({
  shareButtons: {
    display: 'flex',
    flexDirection: 'row',
    color: 'gray',
    cursor: 'pointer',
    marginRight: ''
  },
  photo: {
    marginRight: '10px'
  },
  video: {
    marginRight: '10px'
  },
  docs: {
    marginRight: '50px'
  },
  vl: {
    borderLeft: '1px solid gray',
    height: '30px',
    left: '50%',
    top: 0,
    marginLeft: '10px'
  },
  tooltip: {
    backgroundColor: 'white',
    color: 'grey'
  }
}))
const NewPostFooter = () => {
  const classes = useStyles()
  const longText1 = `Add photo`
  const longText2 = `Add video`
  const longText3 = `Add documents`
  return (
    <>
      <div className={classes.shareButtons}>
        <Tooltip className={classes.tooltip} title={longText1} placement={'top'}>
          <div className={classes.photo}>
            <PhotoSizeSelectActualIcon/>
          </div>
        </Tooltip>
        <Tooltip title={longText2} placement={'top'}>
          <div className={classes.video}>
            <YouTubeIcon/>
          </div>
        </Tooltip>
        <Tooltip title={longText3} placement={'top'}>
          <div className={classes.docs}>
            <EventNoteIcon/>
          </div>
        </Tooltip>
      </div>
      <div className={classes.vl}></div>
    </>
  )
}
export default NewPostFooter