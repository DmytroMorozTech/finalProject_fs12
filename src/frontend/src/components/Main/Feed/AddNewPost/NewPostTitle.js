import React from 'react'
import {makeStyles} from '@material-ui/core/styles'

const useStyles = makeStyles(() => ({
  title: {
    width: '100vh',
    marginLeft: '20px',
    color: 'gray'
  }
}))
const NewPostTitle = () => {
  const classes = useStyles()
  return (
    <div className={classes.title}>
      <h4>Create post</h4>
    </div>
  )
}
export default NewPostTitle