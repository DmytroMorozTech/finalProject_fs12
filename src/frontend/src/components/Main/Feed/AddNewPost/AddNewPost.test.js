import { shallow } from 'enzyme'
import * as redux from 'react-redux'
import React from 'react'
import AddNewPost from './AddNewPost'

jest.mock('./styles', () => () => ({}))

const setUp = () => shallow(<AddNewPost />)

describe('should render AddNewPost component', () => {
  let spyOnUseSelector
  let spyOnUseDispatch
  let mockDispatch

  let component

  beforeEach(() => {
    spyOnUseSelector = jest.spyOn(redux, 'useSelector')
    spyOnUseSelector.mockReturnValue(false)
    spyOnUseDispatch = jest.spyOn(redux, 'useDispatch')
    mockDispatch = jest.fn()
    spyOnUseDispatch.mockReturnValue(mockDispatch)

    component = setUp()
  })
  afterEach(() => {
    jest.restoreAllMocks()
  })
  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
})
