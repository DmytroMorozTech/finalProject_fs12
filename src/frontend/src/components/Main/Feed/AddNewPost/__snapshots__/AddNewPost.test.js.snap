// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddNewPost component to match snapshot 1`] = `
<div>
  <div>
    Create a post
  </div>
  <hr />
  <WithStyles(WithStyles(ForwardRef(DialogContent)))>
    <div>
      <div>
        <Image
          alt="user avatar"
          type="smallAvatar"
        />
      </div>
      <div>
        <WithStyles(ForwardRef(Typography))
          variant="h5"
        />
        <button>
          <div>
            <Memo(ForwardRef(PublicIcon))
              fontSize="inherit"
            />
          </div>
          <WithStyles(ForwardRef(Typography))
            variant="h4"
          >
            Anyone
          </WithStyles(ForwardRef(Typography))>
        </button>
      </div>
    </div>
    <WithStyles(ForwardRef(InputBase))
      fullWidth={true}
      minRows={7}
      multiline={true}
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="What do you want to talk about?"
      value=""
    />
  </WithStyles(WithStyles(ForwardRef(DialogContent)))>
  <div>
    <div>
      <Memo(ForwardRef(RemoveCircleIcon))
        fontSize="inherit"
      />
      <div>
        You have exceeded the maximum character limit.
      </div>
    </div>
    <div>
      3000
    </div>
  </div>
  <WithStyles(WithStyles(ForwardRef(DialogActions)))>
    <div>
      <WithStyles(WithStyles(ForwardRef(Tooltip)))
        placement="top"
        title="Add a photo"
      >
        <div>
          <label>
            <input
              accept="image/*"
              disabled={false}
              id="file"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                }
              }
              type="file"
            />
            <Memo(ForwardRef(PhotoSizeSelectActualIcon)) />
          </label>
        </div>
      </WithStyles(WithStyles(ForwardRef(Tooltip)))>
      <WithStyles(WithStyles(ForwardRef(Tooltip)))
        placement="top"
        title="Add a video"
      >
        <div>
          <label>
            <input
              accept="video/*"
              disabled={false}
              id="file"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                }
              }
              type="file"
            />
            <Memo(ForwardRef(YouTubeIcon)) />
          </label>
        </div>
      </WithStyles(WithStyles(ForwardRef(Tooltip)))>
      <hr />
    </div>
    <SharedButton
      disabled={true}
      onClick={null}
      title="Post"
    />
  </WithStyles(WithStyles(ForwardRef(DialogActions)))>
</div>
`;
